### DISCUSSION TITLE
Ethernet Setup

### DISCUSSION THREAD

**Discussion by sotatech** [gekkekoe/esphome-ecodan-hp#75] [Announcements]
I just got my Ecodan heat pump installed, so I eagerly plugged in the Atom S3 Lite with the modified cable in the instructions. It works!  Thanks for all your work on this.
Sadly, as soon as I refitted the cover on the heat pump, the unit went offline. I suspect that this is due to poor Wi-Fi coverage. 
I do have an Ethernet point beside unit, what would be the best option to get that working? Is there an Ethernet module for the S3 Lite or do I need to get a completely different ESP32 module?
Thanks!

**Comment by F1p** [comment]
I haven't done a config file for the WT32-ETH01 in this repo/project yet, you would also need a FTDI programmer to flash it with ESP home as there is no onboard USB.

You can find a solution on a popular [auction ](https://www.ebay.co.uk/itm/326278295259) site :)

**Reply by sotatech** [reply]
I was hoping to re-use the S3-Lite to simplify configuration, and I've found this: [https://docs.m5stack.com/en/atom/Atomic%20PoE%20Base](url)
It looks like it will work, although I'm not sure about feeding to power from PoE and from the Ecodan at the same time?

**Reply by F1p** [reply]
I saw a post where someone caused damage to the FTC board by feeding 5V in the CN105 rather than drawing power from it.
Like never use the USB port to power the S3-Lite while plugged into the FTC.

It might be better to not use the 5V line, but i think @gekkekoe had other information


You will also need to modify the code to accommodate the PoE base

**Reply by sotatech** [reply]
That was my concern, but if I disconnect the +5V from CN105 and just use the PoE power, how does the grounding work? So I think a non-PoE solution would be safer.
Or maybe just fix my Wi-Fi as @gekkekoe suggests.

**Comment by gekkekoe** [comment]
using that eth port to wire another accesspoint is also an easy solution?

**Reply by sotatech** [reply]
Yes, but where's the challenge in that :)

**Reply by gekkekoe** [reply]
You can use heishamon v5 board, that has built in ethernet and is supported already. Or any esp32 with a supported ethernet chipset (esphome). You then need to configure it like I did for the heishamon board.

@fonske didn't you tried the PoE for the atoms?

But I would use an additional accesspoints.

**Reply by fonske** [reply]
@gekkekoe No, i didnt try PoE.

**Reply by razorclam** [reply]
Continuing this topic, I have one of [these](https://thepihut.com/products/atomic-poe-base-w5500?variant=43133483974851&country=GB&currency=GBP&utm_medium=product_sync&utm_source=google&utm_content=sag_organic&utm_campaign=sag_organic&gad_source=1&gclid=EAIaIQobChMIyqbY8M-JjAMVpJKDBx2X2BquEAQYASABEgIzjvD_BwE)

Will not be using POE so may I ask what's involved to swap from using WIFI to ethernet connection. Is this even possible?

**Reply by gekkekoe** [reply]
> Continuing this topic, I have one of [these](https://thepihut.com/products/atomic-poe-base-w5500?variant=43133483974851&country=GB&currency=GBP&utm_medium=product_sync&utm_source=google&utm_content=sag_organic&utm_campaign=sag_organic&gad_source=1&gclid=EAIaIQobChMIyqbY8M-JjAMVpJKDBx2X2BquEAQYASABEgIzjvD_BwE)
> 
> Will not be using POE so may I ask what's involved to swap from using WIFI to ethernet connection. Is this even possible?

you can have a look at https://github.com/gekkekoe/esphome-ecodan-hp/blob/1b45f529712dab163272c648ec9061d8b615813e/confs/esp32s3-heishamon.yaml#L84

its the same chip, disable the wifi (comment line https://github.com/gekkekoe/esphome-ecodan-hp/blob/1b45f529712dab163272c648ec9061d8b615813e/ecodan-esphome.yaml#L69) and set the correct gpios. see https://docs.m5stack.com/en/atom/Atomic%20PoE%20Base
![image](https://github.com/user-attachments/assets/df14d068-c47a-4aa0-bf78-4eaca9c04021)

the reset and interrupt pin are optional: https://esphome.io/components/ethernet.html 
should be pretty straight forward.
```
# ethernet:
#   type: W5500
#   cs_pin: GPIO6
#   mosi_pin: GPIO8
#   clk_pin: GPIO5
#   miso_pin: GPIO7
```

**Reply by razorclam** [reply]
This is excellent. Thanks so much for your help

**Reply by razorclam** [reply]
Just in case I am over thinking this. I can just copy and paste the ethernet code into the esp32s3.yaml?

**Reply by gekkekoe** [reply]
easiest is your main yaml, but esp32s3.yaml also would work, but then you need to build with local files.

**Reply by razorclam** [reply]
Thanks again. Got it working. Took most of the day!

**Comment by DamijanD** [comment]
@sotatech if I see correctly you have atom S3 inside heat pump? Whole heat pump is more or less from iron and thus this becomes faraday cage (blocking most of EM).
Try with having atom outside of HP - even original wifi module comes with 2m of wire and is meant to be installed outside of the box.

**Comment by gekkekoe** [comment]
@RazorClam Is the ethernet working stable for you ? No random reboots?

**Reply by razorclam** [reply]
Honestly it’s been rock solid since  installed it.

**Reply by gekkekoe** [reply]
what esphome are you running ? Cause I have some one with latest esphome causing some issues. Just trying to see if its introduced since espif 5.3.2
